### ISSUE TITLE
Wi-Fi not connecting after long sleep

### ISSUE THREAD

**Issue by thorro** [syscl/XPS9350-macOS#81]
First of all wow, great work! Almost everything works "out of the box".

I've replaced stock Wi-fi card with DW1560, used the fcvo patch. It's working until computer wakes from long sleep (~8 hours). It sees all the access points, but can't connect (...could not be joined). If I delete the AP and try to reconnect, it asks for password, but then the same error follows.

After reboot it's OK. Deleting them, running the Wifi troubleshooting or changing the Location doesn't help.

Any ideas?

**Comment by ZombieTheBest** [comment]
Same problem. WiFi not connecting or extremely slow after sleep (not sure
if it has to be "long"). This happend since one of the last deploys I think.

Il 20 mar 2017 7:52 AM, "thorro" <notifications@github.com> ha scritto:

**Comment by ZombieTheBest** [comment]
I have DW1830

**Comment by thorro** [comment]
Now that you mention it, I had similar problems after shorter sleep (~20 min). It would connect but ping reply was erratic and took seconds or no reply at all. 

When I connected to another, previously not connected AP, it worked flawlessly, sub 1ms pings.

**Comment by ZombieTheBest** [comment]
@thorro Can confirm it. I didn't have that on previous deploy (or at least I don't think so). 
@syscl Did you changed something regarding Wifi?

**Comment by ZombieTheBest** [comment]
@thorro Fix you provided fixed my issue!

**Comment by thorro** [comment]
@ZombieTheBest the fcvo patch? I've had it all the time, wifi didn't work without it, at all.

In the meantime I've installed the newest Sierra and haven't experienced this problem yes. Hope it stays this way.

**Comment by kiraraso** [comment]
I still have the issue on 10.12.4 with or without the fcvo patch on the DW1830. Wifi is extremely slow/no internet after sleep.

**Comment by furrybanana** [comment]
Hi,
can confirm, the wifi is bugged.

**Comment by syscl** [comment]
Hi guys, can you confirm DW 1560 is also buggy after sleep as well?

I was using Ubuntu 16.04.02 LTS these days due to I have project/due on it, and found DW1830 can even lost connection after waking from sleep. I have had to reboot my laptop to fix this issue on Ubuntu with DW1830.

syscl

**Comment by furrybanana** [comment]
Hi @syscl 
what do I have to change from using 1830 to 1560? I could test this because I have both cards here.

**Comment by furrybanana** [comment]
Hi,
any progress on this Wifi issue (1830)? It was working in February/march before newest macOS and Clover (and latest three deploys, I think).
It is extremely annoying when you let your machine do something while afk and come back after 15 minutes to login again and wifi being extremely slow. Same goes for closing the lid. Would appreciate any ideas to get this fixed.

**Comment by unojazz** [comment]
Same here, extremely slow Internet after sleep

**Comment by linmuzhang** [comment]
hi,syscl,你好，谢谢的XPS9350-macOS 开源项目，就是因为有了这个项目，我才买了xps 9350的，现在有个问题，就是电脑合上进入睡眠状态后，唤醒 后Dw1830 网卡处于关闭状态，即使点击打开也没反应，重启就恢复正常，求解，，测试10.12.2  和 10.12.4都是这样的问题，谢谢

**Comment by Drfrag** [comment]
I think this problem is not related to any changes on deploy.sh but to updates on OS X. 
This issue does not happens on El Capitan using DW1830. And does not happen on Sierra 10.12.5 using DW1560. Something changed on Sierra kexts that bugs DW1830 after sleep. 
With DW1560 I have bluetooth not been available after sleep sometimes.

**Comment by mingiryu** [comment]
@Drfrag 

DW1830 does not have any issue with macOS 10.12.3. It works out of the box and sleep works perfectly. 

I have base model 9350 with data 128GB, so I had to use SATA-100-series-unsupported.kext to boot it up properly. 

I don't think I'll be testing 10.12.5 since it's pretty clear that there is an issue. At the moment, I'm even afraid to open activity monitor->battery because it crashed on 10.12.5 with DW1820A. Then I was never able to bring up activity monitor without reinstalling the OS.

In addition, App Store on 10.12.5 did not work with DW1820A. I'm not sure if it is due to change in version of OS or it's some kind of kernel panic from not using DW1830 when using the deploy.sh script. The App Store works fine if I didn't run the script, but after running it with DW1820A, login and installation does not get verified.

On 10.12.3 with DW1830, wifi works perfectly, App Store works nice, sound, video, keyboard, touchpad all works. Touchpad is somewhat lacking compared to my MacBook Pro. It cannot swipe screen, bring up launchpad, etc. I can swipe screen, tap to click, some-what bring up mission control. Also, It seems that iCloud stuff doesn't work out of the box. I have not been able to test the hardware, so I'm not sure about the USB-C, but regular stuff like usb and charging works OOB. Intel Power Gadget shows that I use about 0.8 W with couple safari tabs open with the SD card disabled. Disable SD card from BIOS makes 0.5W difference.

**Comment by louwe** [comment]
@Drfrag I'm on Sierra and I have a similar problem with my DW1560, but worse.
After a deep sleep (i.e. power button LED turns off), upon wake up, bluetooth will no longer be available, however WiFi still works.
The next deep sleep will then disable WiFi and bring a new issue. The laptop itself will now lose the ability to sleep at all.

This only started happening after I installed the DW1560 card. I had no deep sleep problems prior to swapping DW1820A with DW1560. Of course, the DW1820A never worked at all though.

**Comment by louwe** [comment]
UPDATE:
I may have found the solution to it.
A thought came to me that the newly-installed card may need some "priming" to sleep properly.
So, I rebooted into Windows 10 and did several deep sleep/wake-up cycles and rebooted back to Mac OS Sierra.
To my delight, the priming seems to be the key I needed to solve the problem.

I'm still holding my final judgment though. I will observe to see if the issue crops back up over the next several days.
Stay tuned!

**Comment by brucesong08** [comment]
I have changed dawkwake=10 on clover->boot. It works. Wifi ping value won't be slow.Now I can close my laptop to sleep. 
Noted: Don't use system menu to sleep.

**Comment by louwe** [comment]
UPDATE:
Sleep issue still exists somewhat infrequently. It seems like a solid way to reproduce it is by having a bluetooth device connected while trying to sleep.

Knowing that, I disabled bluetooth in the BIOS and now I have zero sleep issues.
Seems like the Bluetooth card not being able to wake up also directly affects the WiFi card.
Now that it has been disabled though, it can no longer hamper the WiFi card.
That being said, it would still be nice to be able to use the Bluetooth card without fear of the WiFi card being disabled and sleep issues cropping up.

Also, unlike @brucesong08 I didn't have to change darkwake setting so system sleep still works flawlessly. I think I'll live with this solution for now since I don't really care about bluetooth.

**Comment by louwe** [comment]
UPDATE:
I've resolved the issue with some help from folks at tonymacx86.com.

I now have a fully functional WiFi and Bluetooth without any sleep issues whatsoever.
The root cause turns out to be unused USB ports that were still configured.

Since XPS 13 9350 i7-6560U (not sure about other models) only has 2x USB2/3 ports + 3 internal ports, it didn't need the extra 7 ports or so configured by the deploy script listed under SSDT-XHC.aml file (i.e. HS09, USR1, USR2, SS03 - SS06).

I think this should be added together with the WiFi patch from #82 for a glitch-free sleep install.

**Comment by syscl** [comment]
Hello @louwe , nice job, I will look into it and then merge your suggestion!

Thank you, have a great day,
syscl

**Comment by syscl** [comment]
@louwe how about this one?

[SSDT-XHC.aml.zip](https://github.com/syscl/XPS9350-macOS/files/1193266/SSDT-XHC.aml.zip)

Thank you,
syscl

**Comment by syscl** [comment]
Hi guys @thorro @ZombieTheBest @kiraraso @furrybanana @unojazz , please try the latest deploy.sh then report about the Wi-Fi slow issue. 

Thanks in advance,
syscl

**Comment by louwe** [comment]
@syscl I didn't quite use yours (I tweaked my own), but it looks very similar to my configuration.
And yeah it's been a pretty solid week with only 1 reboot so far (to update to OS X 10.12.6).

I did disable BT in the BIOS though. I think my BT problems may stem from the fact that I'm using USB-C port for charging, but for now I'd rather have the USB-C charging over BT connectivity, so I'm fine with it.
